### DISCUSSION TITLE
`packages/tsconfig` in monorepo doesn't allow `"types"` field  in `"compilerOptions"` , is it?

### DISCUSSION THREAD

**Discussion by gyhyfj** [vercel/turborepo#5841] [Help]
Hello, I build a monorepo  with `packages/tsconfig`, like what turbo does, 
However, what's difference, I add `"types"` field  in `"compilerOptions"` like this:
```json [vue.json]
{
  "$schema": "https://json.schemastore.org/tsconfig",
  "extends": "./base",
  "compilerOptions": {
    "types": ["@types/node"]
  }
}
```
and install `@tyoes/node` here, then `package.josn` looked like:
```json [package.json]
{
  "name": "@gyhyfj/ts-config",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "format": "prettier --write ."
  },
  "files": [
    "base.json",
    "vue.json"
  ],
  "dependencies": {
    "@types/node": "latest"
  }
}
```
It works well in dev, but if run `build` script of one of my app which extend that ts config file , it will break down with:
```
error TS2688: Cannot find type definition file for 'node'.
  The file is in the program because:
    Entry point of type library 'node' specified in compilerOptions

Found 1 error.
```
Then I add `"@types/node"` to that app, the `build` script run success again.
Is it normal so I have to install `"@types/node"` in my every app, and add ` "types": ["@types/node"]` to their own `tsconfig.json`, right?

**Comment by anthonyshew** [comment]
There is a lot of "it depends" when it comes to getting something like this right. Could you give us a reproduction to work with?

**Comment by anthonyshew** [comment]
Closing due to staleness.
